airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -97,332 +102,42 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
-                path=lambda self, stream_slice, *args, **kwargs: f"customers/{stream_slice[self.parent_id]}/sources",
-                parent=customers,
-                event_types=["customer.source.created", "customer.source.expiring", "customer.source.updated", "customer.source.deleted"],
-                legacy_cursor_field=None,
-                parent_id="customer_id",
-                sub_items_attr="sources",
-                response_filter={"attr": "object", "value": "bank_account"},
-                extra_request_params={"object": "bank_account"},
-                record_extractor=FilteringRecordExtractor("updated", None, "bank_account"),
-                **args,
-            ),
-            StripeLazySubStream(
-                name="invoice_line_items",
-                path=lambda self, *args, stream_slice, **kwargs: f"invoices/{stream_slice[self.parent_id]}/lines",
-                parent=invoices,
-                parent_id="invoice_id",
-                sub_items_attr="lines",
-                add_parent_id=True,
-                **args,
-            ),
-            subscription_items,
-            StripeSubStream(
-                name="transfer_reversals",
-                path=lambda self, stream_slice, *args, **kwargs: f"transfers/{stream_slice.get('parent', {}).get('id')}/reversals",
-                parent=transfers,
-                **args,
-            ),
-            StripeSubStream(
-                name="usage_records",
-                path=lambda self, stream_slice, *args, **kwargs: f"subscription_items/{stream_slice.get('parent', {}).get('id')}/usage_record_summaries",
-                parent=subscription_items,
-                primary_key=None,
-                **args,
-            ),
+            customers
         ]
         if self._use_concurrent_cdk:
-            return [StreamFacade.create_from_stream(stream, self, entrypoint_logger, 4) for stream in streams]
+            state_manager = ConnectorStateManager(stream_instance_map={s.name: s for s in streams}, state=self._state)
+            return [
+                StreamFacade.create_from_stream(
+                    stream,
+                    self,
+                    entrypoint_logger,
+                    4,
+                    state_manager.get_stream_state(stream.name, stream.namespace),
+                    ConcurrentCursor(
+                        stream.name,
+                        stream.namespace,
+                        state_manager.get_stream_state(stream.name, stream.namespace),
+                        self.message_repository,
+                        state_manager,
+                        CursorField(stream.cursor_field if type(stream.cursor_field) == list else [stream.cursor_field]),
+                        None if state_manager.get_stream_state(stream.name, stream.namespace) else ("created[gte]", "created[lte]"),
+                    ) if self._is_incremental(stream) else NoopCursor(),
+                ) for stream in streams
```

Comment:
Looping on those streams will not be enough as events having a handle on the event creates them in the `__init__` and hence won't be concurrent. It seems like we won't be able to leverage the full capabilities of Concurrent CDK without changing the `source_stripe/streams.py`